src/pages/repositories/index.js

```diff
@@ -0,0 +1,158 @@
+import React, { Component } from "react";
+
+const FILTER_TYPE_MAP = {
+  All: () => true,
+  Public: (repo) => !repo.private,
+  Private: (repo) => repo.private,
+  Fork: (repo) => repo.fork,
+};
+const FILTER_TYPE_NAMES = Object.keys(FILTER_TYPE_MAP);
+
+export default class Repository extends Component {
+  state = {
+    reposData: [],
+    originalReposData: [],
+    repoName: "",
+    repoDescription: "",
+    private: false,
+    fork: false,
+    updateDate: "",
+    isEditMode: false,
+    star: 0,
+    search: "",
+    filterByType: "All",
+    filterByLanguage: "",
+  };
+
+  componentDidMount = () => {
+    this.fetchReposData();
+  };
+
+  fetchReposData = async () => {
+    const response = await fetch("https://api.github.com/users/saritadc/repos");
+    const data = await response.json();
+
+    this.setState({
+      reposData: data,
+      originalReposData: data,
+    });
+  };
+
+  handleOnFilter = (event) => {
+    this.setState({ [event.target.id]: event.target.value });
+  };
+
+  getLanguageName(arrayList, comparisonKey) {
+    const uniqueLanguage = arrayList
```

Comment:
could you please post input arrayList and output uniqueLanguage, just for my reference in the comment please?

Reply:
arraylist is array of all repos:
![image](https://user-images.githubusercontent.com/24495393/120884669-2d78ea80-c604-11eb-9404-77caad996a75.png)

uniqueLanguage is array of repos with each representing different language:
![image](https://user-images.githubusercontent.com/24495393/120884716-8779b000-c604-11eb-84bc-72b68e9b1441.png)